web/js/map-OpenLayers.js

```diff
@@ -349,7 +349,9 @@ $.extend(fixmystreet.utils, {
 
     function replace_query_parameter(qs, id, key) {
         var value = $('#' + id).val();
-        if (value) {
+        if ( $('#' + id).prop('type') == 'checkbox' ) {
+            value = $('#' + id).prop('checked') ? 'on' : '';
```

Comment:
This no longer updates `qs` to put the value in it. I think you want one if/else statement first to set `value`, then another to set/delete `qs[key]`?
Also there seems to be confusion as to whether the value is `1` or `on`. Could you put a `value=1` in the checkbox and then use `1` everywhere without worrying?